doc/adrs/194_improve_diagram_modals.adoc

```diff
@@ -0,0 +1,189 @@
+= ADR-194 - Improve diagram modals
+
+== Context
+
+We have several components, context and custom hooks to handle the current palettes modals.
+
+We also added a new modal in SiriusWeb "manage visibility".
+
+The palette evolved a lot but some old behaviours could be simplified, also the behaviour and style of all modals should be the same.
+
+
+=== How the modal is closed
+
+Currently the palette modal is closed when : 
+
+	- Clicking on the arrow located on the header
+	- Opening another modal
+	- Pressing escape
+	- Changing the selection by selecting another semantic element in the explorer
+
+_note that the palette does not close when clicking outside the xyFlow container_
+
+We actually listen to several events in order to close the palette and not just native events like `onClick` or `onKeyPressed`.
+
+We're also listening to selection changes (this make every component that wants to close the palette to rerender if the selection changes even if the palette is already closed).
+
+We close the palette `onMoveStart`, `onClick`, `onSelectionStart`, `onDragNode`, `onConnectStart`, in `useAdjustSize` and when `DiagramDirectEditInput render`.
+
+By wrapping the palette in a `ClickAwayListener` we could close the palette everytime we click outside the modal.
+
+We would lose the fact that the palette is currently open when clicking outside the xyFlow pane.
+
+But by checking if one of the parent of event.target is the reactFlow renderer we could have the same behaviour as before.
+
+=== How the modal is opened
+
+To open the modal we use a context with a custom hook to handle the state of the modal (open/closed)
+
+This does allow us to open multiple modals but we're currently not reusing any logic.
+
+We then render it inside `rf__wrapper` (palette) or inside the `representation_area` (manage_visibility).
+
+There is a decision that should be made as to what should be the dom element parent of the modal.
+
+There is a dedicated https://reactflow.dev/api-reference/components/viewport-portal[xyFlow component] to render the modal inside the rf_wrapper but the rendered modal is then affected by zooming and we don't want that.
+
+_Note that we currently use a custom portal component to render the palette in rf_wrapper._
+
+It could be better to put the modal in `the representation_area` if we want to reuse it in another representation in the future but there is no evidence that the behaviour of the modal is changing because `rf__wrapper` is the parent.
```

Comment:
Why would we want to reuse any of those in another representation? My main concern with the palette would be for example to have a proper palette for the multi-selection but the palette and manage visibility dialog or using the palette else is nowhere near my radar.